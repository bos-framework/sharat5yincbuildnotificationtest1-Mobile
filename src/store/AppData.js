import { AsyncStorage } from 'react-native';

class AppData {

  getItemForKey = async (key, callback) => {

    try {
      const value = await AsyncStorage.getItem(key);
      if (value !== null) {
        // We have data!!
        callback(true, value);
      }
    } catch (error) {
      // Error retrieving data
      callback(false, error.message);
    }

  };

  setItemForKey = async (key, value, callback) => {

    try {
      await AsyncStorage.setItem(key, value, () => {
        callback(true);
      });
    } catch (error) {
      callback(false);
    }
  }
  
}

export default new AppData();