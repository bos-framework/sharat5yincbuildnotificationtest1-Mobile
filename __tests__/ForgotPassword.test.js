import 'react-native';
import React from 'react';
import ForgotPassword from '../src/Screens/Container/forgotPasswordContainer';
import store from '../src/store/store';


// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer
        .create(<ForgotPassword store={store} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
