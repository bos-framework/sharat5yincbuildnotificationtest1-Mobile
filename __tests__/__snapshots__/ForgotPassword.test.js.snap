// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00000040",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-around",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#000",
              "borderRadius": 10,
              "display": "flex",
              "height": 75,
              "justifyContent": "space-around",
              "width": 75,
            }
          }
        >
          <ActivityIndicator
            animating={false}
            color="#999999"
            hidesWhenStopped={true}
            size="small"
          />
        </View>
      </View>
    </Modal>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.2,
          "justifyContent": "center",
          "marginTop": 10,
        }
      }
    >
      <Image
        resizeMode="center"
        source={
          Object {
            "testUri": "../../../src/Assets/Images/logo.png",
          }
        }
        styles={
          Object {
            "height": 160,
            "width": 160,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 0.8,
          "marginTop": 15,
          "maxWidth": 270,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 14,
            "textAlign": "left",
          }
        }
      >
        Enter the email address associated with your account, and we’ll email you a temporary password.
      </Text>
      <View
        style={
          Array [
            Object {
              "borderColor": "#bdbdbd",
              "borderRadius": 0,
              "borderWidth": 1,
              "marginTop": 10,
            },
            Object {
              "borderColor": "#bdbdbd",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          keyboardType="email-address"
          onChangeText={[Function]}
          placeholder="Email address"
          placeholderTextColor="#919090"
          rejectResponderTermination={true}
          returnKeyType="done"
          secureTextEntry={false}
          style={
            Object {
              "backgroundColor": "#dbdbdb",
              "borderRadius": 0,
              "color": "#545454",
              "fontSize": 16,
              "height": 44,
              "paddingBottom": 0,
              "paddingLeft": 10,
              "paddingTop": 2,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#424142",
            "borderRadius": 2,
            "height": 44,
            "justifyContent": "center",
            "marginTop": 25,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 17,
            }
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
